### DISCUSSION TITLE
Issues with Write Text and Press Keys

### DISCUSSION THREAD

**Discussion by bassjack88** [streamdeck-linux-gui/streamdeck-linux-gui#194] [Q&A] [answered]
Hello! I recently upgraded from Ubuntu 20.04 to 22.04. I was working for a very long time on the streamdeck_ui (unfortunately don't know which version) and had no issues with it. I am only using the Write Text and Press Keys functions as they are sufficient for me.
After upgrading to 22.04 I had no luck with getting streamdeck_ui to work, but after upgrading to some newer version it finally worked. But to my surprise the Press Keys and Write Text were not working in Terminal or Sublime Text, only in Chrome. So stupid me instead of restarting I started digging and finally switched to streamdeck-linux-gui which unfortunately does not work as I expect. 
First of all I can't go back to streamdeck-ui as I get this error:
`streamdeck
Traceback (most recent call last):
  File "/home/*/.local/bin/streamdeck", line 8, in <module>
    sys.exit(start())
  File "/home/*/.local/lib/python3.10/site-packages/streamdeck_ui/gui.py", line 853, in start
    api.open_config(STATE_FILE)
  File "/home/*/.local/lib/python3.10/site-packages/streamdeck_ui/api.py", line 154, in open_config
    raise ValueError("Incompatible version of config file found: " f"{file_version} does not match required version " f"{CONFIG_FILE_VERSION}.")
ValueError: Incompatible version of config file found: 2 does not match required version 1.`
So I reinstalled streamdeck-linux-gui but here's the issue I encouter:
[Screencast from 04.04.2024 10:20:50.webm](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/assets/165995525/ea75c490-7726-4896-b78d-e3b395d63dbf)

I can't use the Write Text function as it simply ignores Capital letters, symbols like : which require shift to input which is really annoying - especially that it was working this way on the streamdeck-ui with no additional work required. I know I can use combinations like shift+2 in the Press Keys to get an @ but I have many links to websites stored as Write Text (can't bookmark in Guest mode) and rewriting them into format like h, t, t, p, s, shift+; and so on is kind of a pain in the ass. 

2nd issue is that now it is very sluggish - writing down a long link using Press Keys takes a very long time - it was instant on the streamdeck-ui. Is there a setting somewhere I can change to make it write/press keys faster? I'm kind of average user so switching to commands or some other method required some advanced programming would be rather hard sorry in advance and thanks in advance!

**Comment by coolapso** [comment]
The reason why you can't go back to the old package is the reason why this fork was created. The maintainer/creator disappeared and dependencies got updated which broke the original package. 

As for the rest, is a consequence of the Shift towards Wayland, Most distros are now on wayland and the trend is to X11 be less and less common, however due the how wayland works some features may still be quite clunky.

**Reply by bassjack88** [reply]
What I discovered now which is very weird but also kinds of solves my issues - in Chrome the Write Text function works as expected - so the sample link https://sampleissue!@#$ is entered accordingly in the url bar :rofl: - but it does not work in Terminal or Sublime Text and other places. 
Ok I can live with that I can change some buttons into Press Keys with shift+2 combinations etc.

But still is there a way to make it faster?

**Comment by bassjack88** [comment, marked as answer]
Ok I found a solution to my issue - in the lib/python3.10/site-packages/streamdeck_ui/modules I found 'keyboard.py' file where I changed the values in line 10 and 11:
_DEFAULT_KEY_PRESS_DELAY = 0.01
_DEFAULT_KEY_SECTION_DELAY = 0.01
changing to those values instead of the default made the write and press keys functions work as it worked previously :) I honestly don't remember if I changed those in Ubuntu 20.04 before I upgraded to 22.04 - I might have probably at the very beginning of using streamdeck 4 years ago, maybe I forgot about it - but if someone else want it faster than it works like a charm now - thanks!
